app/models/concerns/git_hub_plan.rb

```diff
@@ -0,0 +1,28 @@
+module GitHubPlan
+  extend ActiveSupport::Concern
+
+  included do
+    before_validation(on: :create) do
+      verify_organization_has_private_repos_available if private?
+    end
+  end
+
+  def verify_organization_has_private_repos_available
+    created_private_repos = github_assignment_organization.owned_private_repos
+    allowed_private_repos = github_assignment_organization.plan.private_repos
+
+    return if created_private_repos < allowed_private_repos
+
+    error_message = <<-ERROR
+    Cannot make this private assignment, your limit of #{allowed_private_repos}
+    #{'repository'.pluralize(allowed_private_repos)} has been reached
```

Comment:
Cannot make this private assignment, your limit of `#{allowed_private_repos} + #{'repository'.pluralize(allowed_private_repos)}` has been reached. You can request a larger plan for free at https://education.github.com/discount.